### DISCUSSION TITLE
Thinking of considering _My filters_ pane as untrusted by default

### DISCUSSION THREAD

**Discussion by gorhill** [uBlockOrigin/uBlock-issues#2895] [Ideas] [answered]
This would mark as invalid all filters requiring a trusted source.

The workaround to make _My filters_ a trusted source would be to enable `filterAuthorMode` in advanced settings.

The reason is that I worry about users carelessly copy-pasting filters from random sources.

Thoughts?

**Comment by Yuki2718** [comment]
I agree, too many user are copying & pasting rules from obvously untrustful sources now for yt. OTOH this will make debugging filter issues a bit harder when trusted filters are involved as we have to ask him to enable `filterAuthorMode`.

**Comment by ghost** [comment]
Generic procedural filters still are behind a pref, so why not?

**Comment by peace2000** [comment]
I think this is reasonable change. If it causes too much confusion, it can be reverted later (though it's probably fine).

Btw, now as we are speaking about trusted sources, can we also get an advanced option that allows to mark specific lists as trusted? (Something similar like `userResourcesLocation` switch?)

**Comment by gorhill** [comment]
Gave more thoughts to this, taking into account the other requests re. trusted scriptlets.

This will not be controlled by `filterAuthorMode`, I will add a new advanced setting, `trustedListPrefixes`. Possible space-separated values:

- `ublock-`: trust only uBO lists, exclude everything else, including _My filters_ (default value)
- `ublock- user-`: trust uBO lists and _My filters_
- `-`: trust no list, essentially disabling all filters requiring trust (sysadmins or people who don't trust us may want this)

This way, one can also decide to trust lists maintained elsewhere. For example, for stock AdGuard lists add ` adguard-`. Stock EasyList lists, add ` easylist-`.

The token to use are the ones used in [assets.json](https://github.com/gorhill/uBlock/blob/master/assets/assets.json). You fully trust maintainer of List-KR? Add ` KOR-1`.

The matching is made with `startsWith()`, hence why `ublock-` matches all uBO's own filter lists.

So this also opens the door to trust imported lists, for example: add ` https://raw.githubusercontent.com/brave/adblock-lists/master/brave-lists/` to trust all Brave lists. Add ` https://filters.adtidy.org/extension/ublock/filters/` to trust all non-stock AdGuard lists. Use the complete URL of a given list to trust only that one list.

**Reply by ghost** [reply]
`https://` ?

**Reply by gorhill** [reply]
It's an advanced setting, with many obstacle before reaching it. Are you suggesting in case of URL, we should at least limit to a specific domain name?

---

Ok, will limit for now overbroad directives.

**Reply by ghost** [reply, marked as answer]
Yes.

**Comment by partingscientist** [comment]
Considering what people are sharing are filters they're asked to test by a maintainer (which will be outdated by the time it is shared), and the fact that people tend to share these for self-satisfaction [(internet points)](https://old.reddit.com/r/YouShouldKnow/comments/17c90ou/ysk_new_method_to_bypass_youtubes_ad_blocker/), what's stopping them from sharing the entire testing step shared by a maintainer in the future?

I think this is a good change and obviously I don't have any suggestion, but I think I ought to make sure the maintainers know the problem from this angle.

**Reply by gorhill** [reply]
Of course we understand this, and just the same can be said about `userResourcesLocation`, it's not a new concern. The idea is that there are enough steps with warnings on the way.

**Comment by ghost** [comment]
Does this make debugging more difficult when filter maintainer requests user to test some filters? How it's now handled with scriptlets logging available in dev version only?

**Reply by stephenhawk8054** [reply]
For me personally, I usually don't need users testing logging too much (I haven't asked any users about it yet).

**Reply by gorhill** [reply]
Could make _My filters_ trusted by default in dev build?

**Reply by gorhill** [reply]
https://github.com/gorhill/uBlock/commit/e33c0f9d9c0c4cb80169c43caeda7f509bf2a000#diff-25daee2bd4a749c171c5bd0b87567f3fe306a1fdacb2a6d6785a3e1e6fd065f9R92-R94

**Reply by Ai0s** [reply]
@gorhill Could make _My filters_ trusted by default in dev builds. Or add a box at the bottom of the _Support tab_ that holds up to 12 trusted lines for troubleshooting support, that auto clears when the browser is closed or after 10 days.

**Reply by u-RraaLL** [reply]
It already is.  
As for the 2nd idea, it's not gonna happen. Way too much work for something entirely unnecessary.

**Reply by stephenhawk8054** [reply]
Honestly it becomes quite troublesome for me personally as I keep forgetting about that

https://www.reddit.com/r/uBlockOrigin/comments/17tm9rp/youtube_antiadblock_and_ads_november_12_2023_mega/k90bhrw/

Actually I feel not really right to tell users installing dev build just to test one filter.

**Reply by gorhill** [reply]
I wish there were a team of reliable users to help investigate Youtube, surely those who have suffered the warning will have it again when filters stop working. These users would have by then set _My filters_ as a trusted sources. So many people say thanks and want to donate, just helping with investigate filter solutions is the best way to unburden.

**Reply by stephenhawk8054** [reply]
A solution comes to my mind:

- Having a `uBlock filters - Testing` list that's not bundled in uBO but has same update cycle of quick fix list (at least for `1.53`),
- When maintainers need a user to test some trusted filters, tell them to add that list,
- After finishing testing, maintainers will clear the filters -> The filters will disappear after next auto-update on user's side so they won't become major conflicts

Although still quite cumbersome, it's still better than installing dev build / toggling advanced settings + reminding users to be careful about it afterwards IMO.

**Reply by gorhill** [reply]
> same update cycle of quick fix

I rather it has a very long update cycle, and just ask people to click a link to have it updated. We do not want to add more requests to servers than what we will have with diff-update.

**Reply by stephenhawk8054** [reply]
I mean in order to auto delete the filters to prevent future conflicts?

**Reply by gorhill** [reply]
Can't we just ask them to uncheck the list? Sure the size of the list won't be an issue but as it is, quick-fixes generates more than 3 times more requests the the other default lists, so adding another list with same level of request is going to eat into what I was planning for diff-updatable lists.

I was thinking to just have an announcement that we are seeking a team of volunteers to test Youtube filters when needed, i.e. 10-20 people who have show thoroughness when time came to test new filters. If we end up dealing always with the same team of volunteers to trial filters, it should save a lot of time by not having to deal with random people wondering whether they disregarded the instructions.

**Reply by stephenhawk8054** [reply]
> Can't we just ask them to uncheck the list? Sure the size of the list won't be an issue but as it is, quick-fixes generates more than 3 times more requests the the other default lists, so adding another list with same level of request is going to eat into what I was planning for diff-updatable lists.

Ok this will work too.

> I was thinking to just have an announcement that we are seeking a team of volunteers to test Youtube filters when needed, i.e. 10-20 people who have show thoroughness when time came to test new filters.

Actually the current issues in YouTube threads that need testing were not related to anti-adblock (https://github.com/uBlockOrigin/uAssets/issues/20586#issuecomment-1807158573 and https://github.com/uBlockOrigin/uAssets/issues/20586#issuecomment-1807204213) but unusual ones that only a few users encounter due to YouTube testing new things, so a group of testers might not help in these cases and we'll fallback to asking users directly for testing. If it's widespread, we won't need that much testing since it's easier for maintainers to confirm.

**Reply by Yuki2718** [reply]
Having looked the discussion, why can't we make exceptions of trusted scriptlet rules to work even in untrusted environment? I don't think exception of scriptlet filters must be trusted.

**Reply by dportvine** [reply]
>Honestly it becomes quite troublesome for me personally as I keep forgetting about that
>https://www.reddit.com/r/uBlockOrigin/comments/17tm9rp/youtube_antiadblock_and_ads_november_12_2023_mega/k90bhrw/

If the scriptlet is available in a stable version. Isn't that enough: `trustedListPrefixes ublock- user-`?

**Reply by stephenhawk8054** [reply]
@dportvine Yes I know, what we are discussing is the cumbersome procedure to instruct users enabling advanced settings and changing those configurations. And it would be dangerous if they forgot to turn off advanced settings after testing.

All of that just for testing 1 filter.

**Reply by Yuki2718** [reply]
> as I keep forgetting about that

We will eventually get accustomed to, but if this still is too much trouble the change can be reverted, I guess. In the end those who distribute bad rule can instruct user the above. Or possibly a simple checkbox under the warning message with scary comment like "Allow potentially dangerous code to run in My filters" to add `user-` may be good assuming this won't be too much work?

**Reply by gorhill** [reply]
> why can't we make exceptions of trusted scriptlet rules to work even in untrusted environment

You can.

@stephenhawk8054 

We have this mechanism in the logger which allows to create exception filters, and which are cleared when we close the logger.

How about an internal `ublock-support` list, accessible and/or visible only from the _Support_ pane, which would receive filters to try as proposed by maintainers, but which would cease to exist when closing the dashboard or when quitting uBO -- i.e. those filters would never be persisted.

**Reply by stephenhawk8054** [reply]
I actually think the best way is to have a `Testing/Temporary filters` panel (for example between `Support` and `About` panels, based on @Ai0s idea), in which the filters inside would be automatically deleted after some amount of hours based on clients' time. It would still retain `My filters` being untrusted and help testing much easier without complicated prerequisites from users' sides.

But of course it would cause quite a lot of code work that I think better to be considered in next dev cycles.

**Reply by stephenhawk8054** [reply]
@gorhill Oh I didn't realize you have another idea right before my comment. Yes that would work too.

---

Both ideas are quite similar I think, whatever being easier to develop/maintain.

**Reply by Yuki2718** [reply]
> You can.

You mean you are to implment? The current issue is it doesn't: https://github.com/uBlockOrigin/uAssets/issues/20586#issuecomment-1807129637
I confirmed it is not highlited as red in code viewer but anyway doesn't work. ~~Wait.. doesn't work even with `user-` added??~~ [Update] worked after fiddling around, but if without `user-` just adding exception to My filters doesn't work. However, disabling through `#@#` button on the logger works.

**Reply by peace2000** [reply]
>How about an internal ublock-support list, accessible and/or visible only from the Support pane, which would receive filters to try as proposed by maintainers, but which would cease to exist when closing the dashboard or when quitting uBO -- i.e. those filters would never be persisted.

The section that is visible in the Support pane only sounds good.

But maybe with a fixed time limit?

Some users could close the dashboard or even the browser and then test those filters and then claim that they don't work.

**Reply by gorhill** [reply]
> but if without user- just adding exception to My filters doesn't work

Fixed in rc8.

**Reply by stephenhawk8054** [reply]
> How about an internal `ublock-support` list, accessible and/or visible only from the Support pane, which would receive filters to try as proposed by maintainers, but which would cease to exist when closing the dashboard or when quitting uBO -- i.e. those filters would never be persisted.

@gorhill Sorry for the ping. Could this feature be considered in this dev cycle? It would be easier for users to test trusted filters without advanced steps.

A bit modified could be using the timer since the last `Apply changes` for that temp list/panel; for example after 1 hour after the last `Apply changes` button being applied, the filters will be automatically removed in that list. But if quitting uBO/closing browser is easier to implement, it would be good too.

**Reply by gorhill** [reply]
I think it would be simpler if we offer users the choice of trusting the content of _"My filters"_. How about a checkbox at the top of _"My filters"_ pane:

- [ ] Do no reject filters requiring trust

Or however better way to state it in a concise manner.

**Reply by uBlock-user** [reply]
> Or however better way to state it in a concise manner.

How about a checkbox for my filters in filter lists pane near the my filters entry ?

**Reply by gorhill** [reply]
The layout in "Filter lists" pane is dynamically generated and not conducive to add other stuff like this. Anyway, best to have that checkbox right under the warning about copy/pasting filters from untrusted sources.

**Reply by u-RraaLL** [reply]
Sounds good. I'd probably put "I understand the risks." at the beginning though.

**Reply by stephenhawk8054** [reply]
Hmm... I personally still prefer the concept of trusted "My filters" requiring advanced user.

How about the toggle of allowing trusted filters in "My filters" being automatically turned off after restarting browser?

**Reply by Yuki2718** [reply]
Many user will just check the box(es) without second thought.

**Reply by u-RraaLL** [reply]
How about adding `Enable filter tester mode` below `I am an advanced user` in `Settings`?

When ticked, it would enable the toggle in `My filters`. And the mode or the toggle for trusted filters would last until dashboard remained open, similar to temporary exceptions in the logger?

Or would it be possible/better to make use of the temporary filter framework from the logger and add a field there for testing filters instead?

**Reply by Yuki2718** [reply]
I wonder if volatile filters can confuse diagnosis, user may restart browser and unknowingly say the rule we asked doesn't work. Not sure what will be the best, but only putting `Do no reject filters requiring trust` seems too low barrier for misuse.

**Reply by gorhill** [reply]
> trusted "My filters" requiring advanced user

This would require toggling on advanced user mode to test filters, which is an issue if the user has any number of host rules maybe present due to past attempt at using advanced mode.

> toggle of allowing trusted filters in "My filters" being automatically turned off after restarting browser

That sort of unpredictable behavior is user-hostile, nobody would want such behavior -- especially people using the setting wilfully and understanding the risks.

> Many user will just check the box(es) without second thought.

Well the warning is there (including now permanently in mobile), the setting is disabled by default -- at some point we have to yield responsibility for using risky filters to the users. The checkbox just makes it more convenient.

Another idea:

Under uBlockOrigin, we create a new repo to hold ephemeral "lists", when we want a solution to be tried, we create a new "list" with all the filters we want to be tried, then we provide a link for the user to subscribe to the list. Internally, at launch time uBO will discard lists which URL starts with `https://raw.githubusercontent.com/uBlockOrigin/gists/` to ensure proper cleanup of temporary trialed filters -- and at the same time we can `git rm` a list once we get the needed feedback. Forgot to mention, the lists under `https://raw.githubusercontent.com/uBlockOrigin/gists/` would be automatically trusted internally.

**Reply by stephenhawk8054** [reply]
> That sort of unpredictable behavior is user-hostile, nobody would want such behavior -- especially people using the setting wilfully and understanding the risks.

I think it depends on how we interpret that toggle. If we say it like `Temporarily allow trusted filters`, I don't think it will cause that much confusion. Or if we need even clearer, ` - until next browser launch` at the end.

---

A temporary `gist` file means that if user restarts the browser to test, they should subscribe to that list again? And the issue is I don't know how long will that gist file be updated upstream if we modify it so we would be not sure which filter is being applied at user's machine, in case we need to test multiple times.

Actually I would choose a simple toggle as in the first suggestion (without automatic turning-off) rather than the `gist` file approach.

**Reply by gorhill** [reply]
> as the first suggestion

Which one is this?

**Reply by stephenhawk8054** [reply]
> I think it would be simpler if we offer users the choice of trusting the content of _"My filters"_. How about a checkbox at the top of _"My filters"_ pane:
>
> - [ ] Do no reject filters requiring trust
>
> Or however better way to state it in a concise manner.

Oh I mean this one: https://github.com/uBlockOrigin/uBlock-issues/discussions/2895#discussioncomment-8669020

**Reply by Yuki2718** [reply]
> at some point we have to yield responsibility for using risky filters to the users.

Okay, if `Do no reject filters requiring trust` button requires lower workload than ephemeral lists, that will be fine. There are just many more important issues.

**Comment by uBlock-user** [comment]
Could you also make it less restrictive by targeting domains instead of sub-domains for adding imported lists as trusted sources ? Example - `https://pages.dev/` instead of `https://sub-domain.pages.dev/` ?

**Reply by gorhill** [reply]
At this point I prefer to no longer change too much the matching code. Assuming I would agree for such change, asking that `https://sub-domain.pages.dev/` matches `https://pages.dev/` is not trivial, it's no longer a case of simply using `startsWith()`, which currently works well for the majority of cases. Anyways, in which real world case does adding distinct subdomains become an issue?

**Reply by uBlock-user** [reply]
>  in which real world case does adding distinct subdomains become an issue?

https://anti-circumvention-filterlist.pages.dev/ and https://ubo-personal-filters.pages.dev/

**Reply by gorhill** [reply]
That's just two directives, not worth to complicate the code, I rather keep it simple unless a more convincing case is made.

**Reply by uBlock-user** [reply]
Well the issue is of annoyance of having to add the sub-domains individually, so I don't think metric of convincing will apply here, as some people may not find that route annoying at all, but as you said before -  the code changes are not trivial, so I digress.

**Comment by gorhill** [comment]
I want to add a warning sentence at the top of _My filters_, following the one already in there, but I am not confident enough about my English being correct. So I am thinking of:

Do not add filters from untrusted sources.

**Reply by ghost** [reply]
As a native English speaker, I can confirm what you have written is just fine @gorhill.

**Reply by iam-py-test** [reply]
Could there be a link to a Wiki page describing the issue in more depth next to that?
If so, should that explanation be added to the existing My Filters page or a new one?

**Reply by gorhill** [reply]
https://github.com/gorhill/uBlock/commit/f1ce3b2191524a7ee24f1a59c699a67aaf700fad

**Comment by THEtomaso** [comment]
In addition to uBO's native list and 'My filters', I'm also using a large amount of custom filters (all hosted at `https://` locations).
If I've interpreted that new `trustedListPrefixes` setting correctly, this value will allow all of those filters?:
`ublock- user- https://`

**Reply by ghost** [reply]
No https://github.com/uBlockOrigin/uBlock-issues/discussions/2895#discussioncomment-7348147

Domain must match.

**Reply by THEtomaso** [reply]
OK.
Wow, I hope you guys are prepared for the insane amount of issue reports that this setting is likely to cause!? :)

**Reply by THEtomaso** [reply]
I also hope that there's no limit to how many characters the value for `trustedListPrefixes` can hold?

**Reply by gorhill** [reply]
> insane amount of issue reports

Makes no sense. The _"large amount of custom filters (all hosted at https:// locations)"_ are already not allowed to support trusted filters, and so far no insane amount of issue reports because of this.

**Reply by THEtomaso** [reply]
I must be misunderstanding this whole thing then.
What defines "trusted" and "untrused" filters?
All of my custom filters have been working just fine, this far.

**Reply by iam-py-test** [reply]
>  What defines "trusted" and "untrused" filters?

Trusted scriptlets, ie https://github.com/gorhill/uBlock/wiki/Resources-Library#trusted-set-constantjs-
And urltransform: https://github.com/gorhill/uBlock/commit/2e4525fe3c

**Reply by THEtomaso** [reply]
Ah, so it's about rules that loads assets from various sources?
I thought it was about the filter lists themselves!
My bad then. :)

**Reply by gorhill** [reply]
> rules that loads assets from various sources

It's about filters which require to be loaded from a trusted source. All uBO's lists are deemed "trusted source" by default because we control their content here. Using a filter requiring trust in a untrusted source will cause that filter to be rejected.

Using the latest dev build you can already see this in action by viewing the content of "EasyList – Cookie Notices" through the asset viewer (click the eye icon). There is one error reported because it contains a filter which requires to be used in a trusted filter list:

    dokiliko.com##+js(trusted-set-cookie, tracking, 0)

So it is rejected as invalid.

It's of course a typo anyway since the following filter would work given the value used is supported in `set-cookie` scriptlet, which does not require to be used only in a trusted list:

    dokiliko.com##+js(set-cookie, tracking, 0)

Anyway, please read release notes, concept of trusted-source was added in [1.50.0](https://github.com/gorhill/uBlock/releases/tag/1.50.0).

**Reply by gorhill** [reply]
EasyList was never in the trusted lists set. The _only_ change as per discussion here is to remove _My filters_ as trusted source by default. Nothing else. That filter I pointed out in EasyList simply needs to be fixed in EasyList, nothing more.

**Comment by u-RraaLL** [comment]
>> Considering people such as myself that are using widescreen monitors - I'm not sure the plain text warning at the very end of a single line will capture enough attention.
>> 
>> Perhaps font-weight: bold; would be a good choice here?
>
> Originally I was planning to use `color: var(--info3-ink);`, but I feared people might complain it's annoying:
> 
> ![image](https://user-images.githubusercontent.com/585534/277193780-411a54d8-dfe2-4a57-8076-d714c75d677f.png)

Yes, a color might be too distracting in the long run. Plus, it's difficult to choose a color for both themes (brown?).

In any case, some kind of distinction would be better. At the very least `font-style: italic;` might do the trick.

**Reply by gorhill** [reply]
The next build is going out with that change, let's fine tune in the next builds in coming days. I did check with dark theme and it looked fine to me.

**Reply by gorhill** [reply]
If we drop the color, personally I find bold text to be a good choice.

![image](https://github.com/uBlockOrigin/uBlock-issues/assets/585534/a9a2de8c-cb48-4b8f-9df4-b17c134ffd46)

Italic looks terrible on my side.

![image](https://github.com/uBlockOrigin/uBlock-issues/assets/585534/ab9df3de-313d-482b-b8a2-972043013c6a)

**Reply by gorhill** [reply]
Meh, I didn't submit the build to AMO yet, so I will just cancel the new one and ship the bold text change. That should be last changes for today.

**Reply by partingscientist** [reply]
What are your thoughts on placing the warning text just above the "Apply Changes" button? Considering the "Apply Changes" button is often interacted with, the warning text will consequently be read often too, either consciously or subconsciously. At the moment, one would need to read the the whole text line to notice the warning at the end of the description text.

**Reply by gorhill** [reply]
I just did that not long after my last commit:

![image](https://github.com/uBlockOrigin/uBlock-issues/assets/585534/47b20a0b-70ec-432c-bfca-bbd6b4ef68ee)

I decided I will go with this in next release candidate.

**Reply by partingscientist** [reply]
While you're at it, adding a similar warning to the import box in the `Filter lists` tab is probably a good idea, no? I saw someone earlier on the Youtube Reddit thread recommending manually importing `uBO - Annoyances` and people seem to just follow and spread the advice. Today it's uBO's own filters, but it could've been worse in any other day.

**Reply by ghost** [reply]
People are convinced that filters are still just a simple network blocking rules.

**Reply by peace2000** [reply]
By default, imported lists are not trusted but people might add lists that are otherwise bad, they might cause breakages and unnecessary reports at uAssets.

**Reply by MasterKia** [reply]
AdGuard lets users explicitly "trust" a third-party filter list using a GUI option, I don't know why it should be behind a hidden setting in uBO.

**Reply by Yuki2718** [reply]
Actually AG has three trust levels and what corresponds to uBO's trusted list (full trust) is never available to 3p lists.

**Reply by MasterKia** [reply]
But "My filters" is still trusted by default?

**Reply by Yuki2718** [reply]
Yes.

**Reply by THEtomaso** [reply]
So, it's safe to say that setting 'My filters' to untrusted will cause quite a few problems for people unaware of this change, when testing various rules.

**Comment by DandelionSprout** [comment]
I at least give my approval to the Advanced Settings option. After all, there's more than a few scriptlets I want to (or sometimes have to) learn by trial and error, a fair few of which are considered trusted.

**Comment by ghost** [comment]
`replace-node-text` should have alias with `trusted-` prefix (and eventually be replaced). 

Trusted filters should be marked somehow in editor.

**Reply by gorhill** [reply]
https://github.com/gorhill/uBlock/commit/28d92d38f367e8053e6303cad35c4b7790dc5927

**Reply by BlazeFTL** [reply]
So At Some Point All The rpnt Filters Has To Be Re-Edited To trusted-rpnt ?
To Work ?

**Reply by Yuki2718** [reply]
Being alias means no need to do so.

**Reply by gorhill** [reply]
The goal is to have all trusted scriptlets to be prefixed with `trusted-`, and `rpnt` stood out as the only exception. Eventually `rpnt` and `replace-not-text` will be deprecated, so yes, I suggest to re-edit them.

**Reply by BlazeFTL** [reply]
One Minor Suggestions...Trusted- Could Be Shorted To t-rpnt/tr/ts-rpnt...Using Shorter Names Would Be Great

**Reply by gorhill** [reply]
The idea of using `trusted-` is to make it obvious that the scriptlets have special status and to make it easy to find in lists. So `trusted-` it is. There the alias `trusted-rpnt` if you want.

**Reply by krystian3w** [reply]
You could do sc / `trusted-sc`, slsi / `trusted-slsi`, sssi / `trusted-sssi`, `trusted-ce`, sa / `trusted-sa`.

**Comment by Yuki2718** [comment]
BTW shouldn't the checkbox sync with `trustedListPrefixes`?

**Reply by gorhill** [reply]
I guess unchecking should also remove the now obsolete `user-` from the setting.

**Reply by Yuki2718** [reply]
Not the case on my 1.57 on Chrome. It seems if you added `user-`, the unchecking will never be saved.

**Reply by gorhill** [reply]
I meant it should as in it currently does not.

**Comment by D4niloMR** [comment]
So for one to actually add trusted filters from AdGuard, for example: [Tracking Protection filter](https://adguard.com/kb/pt-BR/general/ad-filtering/adguard-filters/)

Have to import `https://raw.githubusercontent.com/AdguardTeam/FiltersRegistry/master/filters/filter_3_Spyware/filter.txt` as a custom list and add the same url to `trustedListPrefixes` in [advanced settings](https://github.com/gorhill/uBlock/wiki/Advanced-settings)

Just adding `adguard-` doesn't work because there is no trusted filter in `https://filters.adtidy.org/extension/ublock/filters/3.txt`

**Reply by garry-ut99** [reply]
Are you sure the first link you provided is suitable and safe to use in uBlock? By looking at uBlock documentation 
https://github.com/gorhill/uBlock/wiki/Dashboard:-Filter-lists#adguard-filter-lists-in-ublock : 

> Given a fact that both uBlock and [AdGuard](https://adguard.com/en/adguard-browser-extension/overview.html), are similiar-purposed blockers and have high compatibility with each other, there are several [AdGuard's filter lists](https://adguard.com/kb/general/ad-filtering/adguard-filters/#adguard-filters) available in uBlock.
> 
> However keep in mind that some of them are slightly modified and adapted by AdGuard for better compatibility with uBlock, as a result, they might be stored separately from each other, for example AdGuard – Ads (Base Filter) list used by AdGuard browser extensions is stored in: [LINK for AdGuard](https://raw.githubusercontent.com/AdguardTeam/FiltersRegistry/master/filters/filter_2_Base/filter.txt), while AdGuard – Ads used in uBlock is stored in: [LINK for uBlock](https://filters.adtidy.org/extension/ublock/filters/2_without_easylist.txt).


It seems that there are different versions of AdGuard lists:
- the ones adapted for uBlock are hosted on URLs which begin with: 
`https://filters.adtidy.org/extension/ublock/filters/`
- the ones adapted for AdGuard are hosted on URLs which begin with: 
`https://raw.githubusercontent.com/AdguardTeam/FiltersRegistry/master/filters/`

**Reply by D4niloMR** [reply]
> Are you sure the first link you provided is suitable and safe to use in uBlock?

They seem as safe as the list for uBO. The filters that uBO should or not execute are inside a pre-parsing directive. Unless there's something else that I'm not aware.

**Reply by garry-ut99** [reply]
In such a case, curious why there is no trusted filters in uBlock version of AdGuard Tracking Protection Filter, maybe intentional for some reason or overlooked.

**Comment by chrcoluk** [comment]
Hi, is this documented anywhere, just updated ublock origin, and I think this is the first time in all the years using it I have become genuinely confused.

The chck box has appeared under my filters, "Allow custom filters requiring trust", I have loads of filters in my filters, most of which I added myself, and I also have some custom lists in the main filter screen, are these now all disabled, or do they still work by default?  If they not working what do I need to do?  googling led me to this page, and also an outdated document page that doesnt mention the new option.

**Reply by MasterKia** [reply]
Your custom filters and lists will still work fine.

Just enable "Allow custom filters requiring trust" if you use any trusted filters or scriptlets.

**Reply by gorhill** [reply]
Using filters requiring trust without _"Allow custom filters requiring trust"_ being checked will cause those filters to be marked as invalid, so if you do not have errors reported in your _"My filters"_ pane, you are fine.

![image](https://github.com/user-attachments/assets/66572e2c-bd5e-4493-b1e0-dd5eab7151a5)

**Comment by u-RraaLL** [comment]
Out of curiosity:

When we disable "My filters" (in filter lists) to test something, then enable it back, the previously enabled "trusted filters" option will stay enabled. Same if  we disable the list inside its pane and then enable it back without navigating elsewhere within that tab (or refreshing it).

But if we disable it inside the pane, then close the tab, navigate to another pane or refresh it, the option gets disabled.

Are the different behaviors by design?

(updated with correct info)

**Reply by gorhill** [reply]
I wouldn't expect the behavior to be different, and I don't see a difference on my side, the "Allow custom filters requiring trust" checkbox stays enabled when I toggle off/on "My filters" in the "Filter lists" pane.

**Reply by u-RraaLL** [reply]
Ok, my bad. It doesn't get disabled in this case.

It gets disabled if you disable my filters inside the pane and navigate to another pane or refresh it though.

**Reply by gorhill** [reply]
https://github.com/gorhill/uBlock/commit/2bb6999e3fcbe296e151e2bdf9ab647ee0f34df5
